src/app/[locale]/dataset/[id]/page.tsx

```diff
@@ -1,12 +1,18 @@
 import {useTranslations} from 'next-intl';
 import {PageHeader, PageTitle} from '@/components/page';
 
-export default function Details() {
+interface DetailsProps {
+  params: {id: string};
+}
+
+export default function Details({params: {id: encodedId}}: DetailsProps) {
```

Comment:
This is a personal preference, but destructuring values in the function signature can make it harder to read (especially if more arguments come in). How about?

```
function Details(props: Props) {
  const encodedId = decodeURIComponent(props.params.id);
}
```

Reply:
Ok. The Airbnb eslint rules I'm used to working with wanted to destructure every object that is more than one deep. So I'm used to that. But I agree that this does not always help readability. So I' m ok with trying something else :)